lib/admin/WP_Auth0_Admin_Advanced.php

```diff
@@ -134,294 +134,233 @@ public function render_passwordless_method() {
-        <?php echo __( '(For example: {"disableResetAction": true }) ', 'wp-auth0' ); ?>
-      </span>
-    </div>
-    <?php
+  public function render_extra_conf( $args ) {
+    $this->render_textarea_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Valid JSON for Lock options configuration; will override all options set elsewhere. ', 'wp-auth0' ) .
+      $this->get_docs_link( 'libraries/lock/customization', 'See options and examples' )
+    );
   }
 
-  public function render_custom_signup_fields() {
-    $v = $this->options->get( 'custom_signup_fields' );
-?>
-
-    <textarea name="<?php echo $this->options->get_options_name(); ?>[custom_signup_fields]" id="wpa0_custom_signup_fields"><?php echo esc_attr( $v ); ?></textarea>
-    <div class="subelement">
-      <span class="description">
-        <?php echo __( 'This field is the Json that describes the custom signup fields for lock. It should be a valida json and allows the use of functions (for validation). More info', 'wp-auth0' ); ?>
-        <a target="_blank" href="https://auth0.com/docs/libraries/lock/v10/new-features#custom-sign-up-fields"><?php echo __( 'here', 'wp-auth0' ); ?></a>
-
-        <code><pre>[
-  {
-    name: "address",                              // required
-    placeholder: "enter your address",            // required
-    icon: "https://example.com/address_icon.png", // optional
-    prefill: "street 123",                        // optional
-    validator: function(value) {                  // optional
-      // only accept addresses with more than 10 chars
-      return value.length > 10;
-    }
-  },
-  {
-    ... // more fields could be specified
-  }
-]</pre></code>
-      </span>
-    </div>
-    <?php
+  public function render_custom_signup_fields( $args ) {
+    $this->render_textarea_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Valid JSON for custom signup fields in the Auth0 signup form. ', 'wp-auth0' ) .
+      $this->get_docs_link(
+        'libraries/lock/v11/configuration#additionalsignupfields-array-',
+        __( 'More information and examples', 'wp-auth0' )
+      )
+    );
   }
 
-  public function render_link_auth0_users() {
-    $v = $this->options->get( 'link_auth0_users' );
-
-    echo $this->render_a0_switch( "wpa0_link_auth0_users", "link_auth0_users", 1, ! empty( $v ) );
-?>
-      <div class="subelement">
-        <span class="description"><?php echo __( 'Links accounts with the same e-mail address. It will only occur if both e-mails are previously verified.', 'wp-auth0' ); ?></span>
-      </div>
-    <?php
+  public function render_link_auth0_users( $args ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Links accounts with the same e-mail address (emails must be verified)', 'wp-auth0' )
```

Comment:
If possible, add a small sentence telling when an email is considered verified. Such as: "An email is verified when the user was created using a social provider that shared the account's email or when a user receives a verification email and proceeds with the verify flow."

Reply:
Added this to the "Require Verified Email" setting and mentioned it here.